kratos/sources/model_part.cpp

```diff
@@ -855,6 +855,291 @@ void ModelPart::RemoveElementsFromAllLevels(Flags identifier_flag)
 }
 
 
+/*
+    Functions for Master-Slave Constraint
+*/
+
+/** Inserts a master-slave constraint in the current mesh.
+ */
+void ModelPart::AddMasterSlaveConstraint(ModelPart::MasterSlaveConstraintType::Pointer pNewMasterSlaveConstraint)
+{
+    if (IsSubModelPart())
+    {
+        mpParentModelPart->AddMasterSlaveConstraint(pNewMasterSlaveConstraint);
```

Comment:
it should also add it here, not just to the parent. (If you add to a submodelpart it belongs to the subpart AND to the parent)

Reply:
done